cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -783,48 +804,80 @@ private void configProviders(KafkaConfiguration userConfig)    {
-     * @param injectCcMetricsReporter   Inject the Cruise Control Metrics Reporter into the configuration
+     * @param userConfig                     The User configuration - Kafka broker configuration options specified by the user in the Kafka custom resource
+     * @param injectCcMetricsReporter        Inject the Cruise Control Metrics Reporter into the configuration
+     * @param injectStrimziMetricsReporter   Inject the Strimzi Metrics Reporter into the configuration
      *
      * @return Returns the builder instance
      */
-    public KafkaBrokerConfigurationBuilder withUserConfiguration(KafkaConfiguration userConfig, boolean injectCcMetricsReporter)  {
-        if (userConfig != null && !userConfig.getConfiguration().isEmpty()) {
-            // We have to create a copy of the configuration before we modify it
-            userConfig = new KafkaConfiguration(userConfig);
-
-            // Configure the configuration providers => we have to inject the Strimzi ones
-            configProviders(userConfig);
-
-            if (injectCcMetricsReporter)  {
-                // We configure the Cruise Control Metrics Reporter is needed
-                if (userConfig.getConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD) != null) {
-                    if (!userConfig.getConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD).contains(CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER)) {
-                        userConfig.setConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD, userConfig.getConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD) + "," + CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
-                    }
-                } else {
-                    userConfig.setConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD, CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
-                }
-            }
-
-            printSectionHeader("User provided configuration");
+    public KafkaBrokerConfigurationBuilder withUserConfiguration(KafkaConfiguration userConfig,
+                                                                 boolean injectCcMetricsReporter,
+                                                                 boolean injectStrimziMetricsReporter) {
+        // we have to create a copy of the configuration before we modify it
+        userConfig = userConfig != null
+                ? new KafkaConfiguration(userConfig)
+                : new KafkaConfiguration(reconciliation, new ArrayList<>());
+
+        // configure config.providers
+        addConfigProviders(userConfig);
+
+        // configure metric.reporters
+        maybeAddMetricReporters(userConfig, injectCcMetricsReporter, injectStrimziMetricsReporter);
```

Comment:
Why do we have `maybeAddMetricReporters` and `maybeAddKafkaMetricsReporters`?

Reply:
We need to have something to handle the configuration logic. This seems like the best way as part of the overall refactoring.

Reply:
My question is more why do we need these two methods and what are the differences? I assume we need at least one, but not sure why we need two - and the names at least for me do not properly explain the difference. I'm not necessarily saying this is wrong ... but I would like to understand it in case you think they are both needed.

Reply:
There are two methods because `maybeAddMetricReporters` is for kafka metrics and `maybeAddKafkaMetricsReporters` is for yammer metrics. `maybeAddMetricReporters()` adds Cruise Control and/or Strimzi Prometheus reporters to metric.reporters.
`maybeAddKafkaMetricsReporters()` adds the Yammer Prometheus reporter to kafka.metrics.reporters.
CC is not part of the kafka.metrics.reporters (Yammer).
We would like to keep them nice and separated. Does this make sense and answer the question?

Reply:
Then perhaps you should make it clear from the naming what does what ... for example `maybeAddYammerMetricsReporters`. You can also utilize comments -> both in the place where you call the methods as well as the Javadocs on the actual methods to make things more clear.

Reply:
Added a Javadoc and comment outlining the difference between the two methods.